Danki2/Assets/Scripts/Camera/CameraShakeManager.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+public class CameraShakeManager
+{
+    private CameraShake[] camShakes;
+
+    public CameraShakeManager(int poolSize)
+    {
+        camShakes = new CameraShake[poolSize];
+
+        for (int i = 0; i < poolSize; i++)
+        {
+            camShakes[i] = new CameraShake();
+        }
+    }
+
+    public Vector3 GetShakeVector()
+    {
+        float maxStrength = float.MinValue;
+        CameraShake strongestShake = default;
+
+        foreach (CameraShake cs in camShakes)
+        {
+            if (cs.Duration > 0f && cs.Strength > maxStrength)
+            {
+                maxStrength = cs.Strength;
+                strongestShake = cs;
+            }
+        }
+
+        if (strongestShake != default)
+        {
+            return strongestShake.GetShakeVector();
```

Comment:
This calls through to `TickDuration()` on the `CameraShake`, so only the strongest camera shake will get ticked, is this what we want? It means that if 2 shakes, lasting 1 second each, get added at the same time, the camera will shake for 2 seconds. It could get a bit nuts with lots of things shaking the camera. I think it'd make more sense to tick all of the shakes

Reply:
Oops - it was definitely the intention to tick all shakes.